### PR TITLE
Refs [TASK][36590] Data HomeScreen

### PR DESCRIPTION
**Pull request by manhhung23399**
## Related Tickets
- [#Task 36590: Data HomeScreen](https://edu-redmine.sun-asterisk.vn/issues/36590)
## WHAT
- bind data recyclerView 3 fragment platform
## Evidence (Screenshot or Video)
image

## Review Checklist

Category | View Point | Description | Expected Reviewer Answer | Self review | Reviewer2 (name)
--- | --- | --- | --- | --- | ---
Conventions | Does the code follow Sun* coding style and coding conventions? | https://github.com/framgia/coding-standards/tree/master/eng/android | YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Redmine | Does the ticket follow Sun* Redmine working process?  | https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md| YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Documentation | Is there any incomplete code? If so, should it be removed or flagged with a suitable marker like ‘TODO’? |  | YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: app/src/main/java/com/sun/gamevui/ui/PcFragment.kt [hunk 1]
```diff
@@ -1,17 +1,25 @@
...
 import org.koin.androidx.viewmodel.ext.android.sharedViewModel
+import java.util.*
 
 class PcFragment : BaseFragment<FragmentPlatformBinding>() {
 
     private val popularAdapter = PopularAdapter(this::clickItemPopular)
+    private val gamesAdapter = GamesAdapter(this::clickItemGame)
     private val genreAdapter by lazy { context?.let { GenreAdapter(it) } }
+    private var genre: Genre? = null
     override val layoutId = R.layout.fragment_platform
```
Comment (phucsaker99) [inline app/src/main/java/com/sun/gamevui/ui/PcFragment.kt]: var vs val tách riêng ra nha cho đỡ nhìn

File: app/src/main/java/com/sun/gamevui/ui/adapter/GamesAdapter.kt [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+                false
+            ), onClickItem
+        )
+
+    class GameViewHolder(
+        private val itemGameBinding: ItemGameBinding,
+        onClickItem: (Game) -> Unit
+    ) : BaseViewHolder<Game>(itemGameBinding, onClickItem) {
+        override fun bindData(item: Game) {
+            super.bindData(item)
```
Comment (phucsaker99) [inline app/src/main/java/com/sun/gamevui/ui/adapter/GamesAdapter.kt]:
```suggestion
    ) : BaseViewHolder<Game>(itemGameBinding, onClickItem) {
    
        override fun bindData(item: Game) {
            super.bindData(item)
```

File: app/src/main/java/com/sun/gamevui/data/model/Game.kt [hunk 3]
```diff
@@ -23,7 +23,7 @@ data class Game(
     val score: Long,
     @ColumnInfo(name = "platforms")
     @SerializedName("parent_platforms")
-    val platforms: List<Platform>
+    val parent_platforms: List<Platforms>
```
Comment (huydd-2678) [inline app/src/main/java/com/sun/gamevui/data/model/Game.kt]: tên sai format này ô

File: app/src/main/java/com/sun/gamevui/ui/PcFragment.kt [hunk 4]
```diff
@@ -1,17 +1,25 @@
...
     override val viewModel by sharedViewModel<HomeViewModel>()
```
Comment (huydd-2678) [inline app/src/main/java/com/sun/gamevui/ui/PcFragment.kt]: thứ tự xếp theo convention ô ơi

File: app/src/main/res/layout/item_game.xml [hunk 5]
```diff
@@ -18,7 +18,7 @@
         app:cardCornerRadius="@dimen/dp_8">
 
         <androidx.constraintlayout.widget.ConstraintLayout
-            android:layout_width="@dimen/dp_200"
+            android:layout_width="@dimen/dp_170"
             android:layout_height="@dimen/dp_200">
```
Comment (huydd-2678) [inline app/src/main/res/layout/item_game.xml]: Fix cứng size à ô?

Comment (manhhung23399) [reply app/src/main/res/layout/item_game.xml]: cái này tôi đặt width height cho layout item

File: app/src/main/res/layout/item_game.xml [hunk 6]
```diff
@@ -78,6 +84,8 @@
                 android:layout_width="0dp"
                 android:layout_height="wrap_content"
                 android:layout_margin="@dimen/dp_4"
+                android:text="@{Long.toString(game.score)}"
```
Comment (huytq-2422) [inline app/src/main/res/layout/item_game.xml]:
```suggestion
                android:text="@{String.valueOf(game.score)}"
```

File: app/src/main/java/com/sun/gamevui/ui/adapter/GamesAdapter.kt [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+            item.parentPlatforms.forEach { platforms ->
+                when (platforms.platform.name) {
+                    "PC" -> itemGameBinding.imagePc.visibility =
+                        View.VISIBLE
+                    "PlayStation" -> itemGameBinding.imagePs.visibility =
+                        View.VISIBLE
+                    "Xbox" -> itemGameBinding.imageXbox.visibility =
+                        View.VISIBLE
+                }
+            }
```
Comment (huytq-2422) [inline app/src/main/java/com/sun/gamevui/ui/adapter/GamesAdapter.kt]: bind luôn trong xml

File: app/src/main/java/com/sun/gamevui/ui/adapter/GamesAdapter.kt [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+import android.view.ViewGroup
+import androidx.databinding.DataBindingUtil
+import com.sun.gamevui.R
+import com.sun.gamevui.base.BaseAdapter
+import com.sun.gamevui.base.BaseViewHolder
+import com.sun.gamevui.data.model.Game
+import com.sun.gamevui.databinding.ItemGameBinding
+
+class GamesAdapter(
+    private val onClickItem: (Game) -> Unit
```
Comment (thonc-2135) [inline app/src/main/java/com/sun/gamevui/ui/adapter/GamesAdapter.kt]:
```suggestion
    private val onItemClick: (Game) -> Unit
```
để ý xem mấy hàm của Android đều viết theo kiểu v, onItemSelected các kiểu chứ có ai viết ngc lại đâu

**Pull request merged**